### ISSUE TITLE
Legend get cut off

### ISSUE THREAD

**Issue by praveenbroski** [PhilJay/MPAndroidChart#2923]
Hi @PhilJay I've used your library to draw line graphs in my android project. Everything works fine. Thanks for the wonderful library.

I just have one problem with the legend. It's getting cut off some of legend labels not visible. I've used this library in iOS too. In swift it's displaying one by one without any trouble. I've referred issues section on this repository and recommended to use setWordWrapEnabled(true). I've tried that still getting my legends cut off.

Here I posted a screen shot. So please kindly provide a way to correct the issue. Thanks
![screenshot_2017-03-16-10-35-51-259](https://cloud.githubusercontent.com/assets/14329980/23982718/3c22e106-0a35-11e7-8583-8dcf6b389c37.jpeg)

**Comment by iammert** [comment]
@praveenbroski Did you find any solution ?

**Comment by praveenbroski** [comment]
Yes. I solved it by putting a gridview below the graph and populated it with legends

**Comment by cadifilippo** [comment]
I could explain how I solved it, with the code, I have the same error

**Comment by praveenbroski** [comment]
I use custom **ExpandableGridView** to achieve my task.

Create the below class in your project.

**ExpandableGridView.java**

`import android.content.Context;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.widget.GridView;


public class ExpandableGridView extends GridView {

    boolean expanded = false;

    public ExpandableGridView(Context context) {
        super(context);
    }

    public ExpandableGridView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public ExpandableGridView(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
    }

    public boolean isExpanded() {
        return expanded;
    }

    @Override
    public void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        // HACK! TAKE THAT ANDROID!
        if (isExpanded()) {
            // Calculate entire height by providing a very large height hint.
            // View.MEASURED_SIZE_MASK represents the largest height possible.
            int expandSpec = MeasureSpec.makeMeasureSpec(MEASURED_SIZE_MASK,
                    MeasureSpec.AT_MOST);
            super.onMeasure(widthMeasureSpec, expandSpec);

            ViewGroup.LayoutParams params = getLayoutParams();
            params.height = getMeasuredHeight();
        } else {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        }
    }

    public void setExpanded(boolean expanded) {
        this.expanded = expanded;
    }

}`

Add ExpnadableGridView widget below your LineChart in xml Layout

          `<com.github.mikephil.charting.charts.LineChart
                    android:id="@+id/mChart"
                    android:layout_width="match_parent"
                    android:layout_height="250dp"
                    android:layout_marginEnd="5dp"
                    android:layout_marginLeft="5dp"
                    android:layout_marginRight="5dp"
                    android:layout_marginStart="5dp"
                    android:layout_marginTop="10dp" />

          <com.domain.ExpandableGridView
                    android:id="@+id/m_legend"
                    android:layout_width="match_parent"
                    android:layout_height="wrap_content"
                    android:layout_margin="10dp"
                    android:isScrollContainer="false"
                    android:listSelector="#00000000"
                    android:numColumns="3"
                    android:stretchMode="columnWidth" />`

Disable default legends of graph by using

`Legend lMc = graph.getLegend();
 lMc.setEnabled(false);`

After loading data to graph use the following method to populate the ExpandableGridView.

Call this using `set_legend(lMc, grid);`

Here lMc is the legend of the graph. I used multiple graphs so i pass the corresponding GridView to the method.

I saved colors in a array in `strings.xml`

`private void set_legend(Legend legend, ExpandableGridView lGrid) {
        List<String> list = new ArrayList<>();
        final String[] colorArray = getResources().getStringArray(R.array.colorList);
        List<String> colors = Arrays.asList(colorArray);
        for (LegendEntry entry: legend.getEntries()) {
            list.add(entry.label);
        }
        LegendAdapter adapter = new LegendAdapter(list, colors);
        lGrid.setAdapter(adapter);
}`

After this you can customize the legend using `BaseAdapter`.

The LegendAdapter is 

**LegendAdapter.java**

`private class LegendAdapter extends BaseAdapter {
        List<String> list = new ArrayList<>();
        List<String> colors = new ArrayList<>();
        LayoutInflater inflater;
        LegendAdapter(List<String> list, List<String> colors) {
            this.list = list;
            this.colors = colors;
            inflater = (LayoutInflater) getSystemService(LAYOUT_INFLATER_SERVICE);
        }

        @Override
        public int getCount() {
            return list.size();
        }

        @Override
        public Object getItem(int position) {
            return position;
        }

        @Override
        public long getItemId(int position) {
            return position;
        }

        @Override
        public View getView(int position, View convertView, ViewGroup parent) {
            convertView = inflater.inflate(R.layout.legend_item, parent, false);

            View view = convertView.findViewById(R.id.view);
            TextView tv = convertView.findViewById(R.id.textView44);
            String entry = list.get(position);
            tv.setText(entry);
            if (entry.equals("Refill")) {
                view.setBackgroundColor(Color.parseColor("#0000FF"));
            } else {
                view.setBackgroundColor(Color.parseColor("#" + colors.get(position)));
            }

            return convertView;
        }
    }`

Please try the above and reply me if any errors.

Happy coding.
